syntax = "proto3";

package todo;

message Todo {
	int32 id = 1;
	string name = 2;
	bool done = 3;
}

message CreateTodoRequest {
	string name = 1;
	bool done = 2;
}

message CreateTodoResponse {
	Todo todo = 1;
}

message GetTodoRequest {
	int32 id = 1;
}

message GetTodoResponse {
	Todo todo = 1;
}

message UpdateTodoRequest {
	int32 id = 1;
	string name = 2;
	bool done = 3;
}

message UpdateTodoResponse {
	Todo todo = 1;
}

message DeleteTodoRequest {
	int32 id = 1;
}

message DeleteTodoResponse {
	bool success = 1;
}

message ListTodosRequest {}

message ListTodosResponse {
	repeated Todo todos = 1;
}

service TodoService {
	rpc CreateTodo(CreateTodoRequest) returns (CreateTodoResponse);
	rpc GetTodo(GetTodoRequest) returns (GetTodoResponse);
	rpc UpdateTodo(UpdateTodoRequest) returns (UpdateTodoResponse);
	rpc DeleteTodo(DeleteTodoRequest) returns (DeleteTodoResponse);
	rpc ListTodos(ListTodosRequest) returns (ListTodosResponse);
}

